report-ng/app/src/components/classes/classes.ts

```diff
@@ -229,20 +318,110 @@ export class Classes extends AbstractViewModel {
     }
 
     private _statusChanged() {
-        this._filterOnce();
+        this._addFilter(this._selectedStatus, ChipType.STATUS);
+        this._selectedStatus = undefined;
     }
 
     private _classChanged() {
-        this._filterOnce();
-    }
+        // remove selected classes from options in select box
+        const index = this._filteredClassStatistics.map(stat => stat.classIdentifier).indexOf(this._selectedClass, 0);
+        if (index > -1) {
+            this._filteredClassStatistics.splice(index, 1);
+        }
 
+        this._addFilter(this._selectedClass, ChipType.CLASS);
+        this._selectedClass = undefined;
+    }
 
     private _showConfigurationChanged() {
         this._filterOnce();
     }
 
-    private _searchQueryChanged() {
+    private _searchQueryChanged($event) {
+        if($event.key == "Enter"){
+            this._addFilter(this._searchInput, ChipType.CUSTOM_TEXT);
+            this._searchInput = undefined;
+        }
+    }
+
+    private _addFilter(element, chipType: ChipType){
+        if(element){
+            if(this._chipList.length <= 0){     // if it is the first chip to appear, the "Clear All" chip has to appear as well
+                const chip = {
+                    chipType: ChipType.CLEAR_ALL,
+                }
+                this._chipList.unshift(chip);
+            }
+            if(!this._chipList.filter(chip => chip.chipType === chipType).map(chip => chip.chipElement).includes(element)){
+                const chip: IChip = {
+                    chipType: chipType,
+                    chipElement: element
+                }
+                this._chipList.push(chip);
+            }
+        }
+
         this._filterOnce();
     }
+
+    private _removeFilter(chipType: ChipType, filterObject){
+        if(chipType == ChipType.CLEAR_ALL){
```

Comment:
I don't like the idea of using semantical "filter chips" that represent the filter state, and reuse them as actions like "clear all filters", that should done by a button instead.

Add a button instead in HTML:
```html
<mdc-button if.bind="_chips.length > 0" click.delegeate="_clearAllFilters()">Clear all filters</mdc-button>
```